src/awkward/_v2/cpp-headers/awkward/LayoutBuilder.h

```diff
@@ -0,0 +1,1706 @@
+// BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+#ifndef AWKWARD_LAYOUTBUILDER_H_
+#define AWKWARD_LAYOUTBUILDER_H_
+
+#include "awkward/GrowableBuffer.h"
+#include "awkward/utils.h"
+
+#include <map>
+#include <algorithm>
+#include <tuple>
+#include <string>
+
+namespace awkward {
+
+  namespace LayoutBuilder {
+
+  template<std::size_t ENUM, typename BUILDER>
+  class Field {
+  public:
+    using Builder = BUILDER;
+
+    std::string index_as_field() const {
+      return std::to_string(static_cast<int>(index));
+    }
+
+    template<typename PRIMITIVE>
+    void
+    append(PRIMITIVE x) {
+      builder.append(x);
+    }
```

Comment:
This method assumes that `Builder::append` exists, but that's not true of all builders. It would fail to compile if a record field's type is ever a list, for instance.

If this is a convenience method (not the only way to do it), then it should be removed.

I think it's not the only way to do it, since `builder` is public.